databricks_blueprints/download_file_from_dbfs.py

```diff
@@ -0,0 +1,132 @@
+    if ('.csv' in source_file_path) or ('.txt' in source_file_path):
+        is_binary = False
+        write_mode = "w+"
+    else:
+        is_binary = True
+        write_mode = 'wb+'
+    with open(dest_file_path, write_mode) as file:
+        offset = 0
+        while True:
+            payload = {
+                "path": source_file_path,
+                "offset": offset, 
+                "length": ONE_MB # 1MB of data 
+            }
+            try:
+                read_response = client.get(read_handle_endpoint, params=payload)
+            except Exception:
+                print(f'Connection Error for {read_handle_endpoint}')
+                print(f"Data downloaded: {offset/1024}MB")
+                sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+
+            if read_response.status_code == requests.codes.ok:
+                response_data = read_response.json()
+                data = base64.b64decode(response_data['data'])
+                if is_binary:
+                    file.write(data)
+                else:
+                    file.write(data.decode('utf-8'))
+                # check if there is still more data left to read
+                if response_data['bytes_read'] == ONE_MB:
+                    offset += ONE_MB
+                else: # that was the last of the data in the file
+                    break
+            elif read_response.status_code == 404:
+                print(f"Error: No file located at {source_file_path}")
+                sys.exit(errors.EXIT_CODE_DBFS_INVALID_FILEPATH)
+            elif read_response.status_code == 400: # Bad Request
+                error = read_response.json()
+                code = error['error_code']
+                message = error['message']
+                print(f"Error: {code} {message}")
+                sys.exit(errors.EXIT_CODE_DBFS_READ_ERROR)
+            else:
+                print(f"Failed to read data: {read_response.status_code} {read_response.text}")
+                sys.exit(errors.EXIT_CODE_DBFS_READ_ERROR)
+    print(f"finished downloading file:{source_file_path} as {dest_file_path}")
+    
+
+def main():
+    args = get_args()
+    access_token = args.access_token
+    instance_id = args.instance_id
+    source_file_name = args.source_file_name
+    source_folder_name = args.source_folder_name
+    dest_file_name = args.dest_file_name
+    dest_folder_name = args.dest_folder_name
+    source_file_name_match_type = args.source_file_name_match_type
+    # create client
+    client = helpers.DatabricksClient(access_token, instance_id)
+    
+    if not source_folder_name:
+        source_folder_name = '/FileStore/'
+
+    if not dest_file_name:
+        dest_file_name = source_file_name
+
+    if not dest_folder_name:
+        dest_folder_name = os.getcwd()
+        
+    # get list of all potential file matches
+    if source_file_name_match_type == 'regex_match':
+        files = helpers.list_dbfs_files(client, source_folder_name)
```

Comment:
When 0 matches are found, it still says "downloading" and results in an exit code of 0. Need to error handle for that.